icaruscode/CRT/CRTTPCTruthEffMCKalman_module.cc

```diff
@@ -0,0 +1,1721 @@
+	art::FindManyP<recob::Hit> findManyHits(trackListHandle, event, trackLabel);
+	
+	// will create art pointers to the new T0 objects:
+//	art::PtrMaker<anab::T0> const makeT0Ptr{ event };
+
+	// Loop over all the reconstructed tracks 
+	for(size_t track_i = 0; track_i < trackList.size(); track_i++) {
+
+	  //clear out vectors for new entries into the TTree
+	  all_crt_candidate_trueIDs.clear();
+	  all_crt_candidate_motherIDs.clear();
+	  all_crt_candidate_ancestorIDs.clear();
+	  all_crt_regions.clear();
+	  all_crt_pdg.clear();
+	  all_crt_candidate_motherlayers.clear();
+	  all_crt_candidate_x.clear();
+	  all_crt_candidate_y.clear();
+	  all_crt_candidate_z.clear();
+	  crt_start_dca.clear();
+	  crt_end_dca.clear();
+	  crt_timestamp.clear();
+	  all_crt_hitcode_map.clear();
+	  all_crt_mac5s.clear();
+	  all_crt_chans.clear();
+	  all_crt_hitcode.clear();
+	  all_crt_pes.clear();
+	  matched_hit_mac5.clear();
+	  matched_hit_chan.clear();
+	  matched_hit_pes.clear();
+	  all_crt_hitflags.clear();
+	  track_points_x.clear();
+	  track_points_y.clear();
+	  track_points_z.clear();
+
+	  //Set certain output variables to defaults
+	  crt_tpc_dca = DBL_MAX; crt_trueID = INT_MAX; 
+	  track_trueID = INT_MAX; track_mother = INT_MAX; track_ancestor = INT_MAX; track_motherlayers = INT_MAX;
+	  crt_trueID = INT_MAX; crt_mother = INT_MAX; crt_ancestor = INT_MAX; crt_motherlayers = INT_MAX;
+	  crttime = DBL_MAX; track_t0=DBL_MAX;
+	  best_dca_pos=-99999; track_length=-99999;
+	  tpc_start_x = DBL_MAX; tpc_start_y = DBL_MAX; tpc_start_z = DBL_MAX;
+	  tpc_end_x = DBL_MAX; tpc_end_y = DBL_MAX; tpc_end_z = DBL_MAX;
+	  tpc_mintick = -99999; tpc_maxtick = -99999;
+	  t0min = DBL_MAX; t0max = DBL_MAX;
+	  startflag=-1; endflag=-1;	
+	  PCA_trackstart_reg1 = -1; PCA_trackstart_reg2 = -1; PCA_trackstart_reg3 = -1;
+	  PCA_trackend_reg1 = -1; PCA_trackend_reg2 = -1; PCA_trackend_reg3 = -1;
+	  tpcEast=false; tpcWest=false;
+	  trk_pdg = -999999; crt_pdg = -999999; trk_ancestor_pdg = -1111111; crt_ancestor_pdg = -1111111;
+	  pfp_id = -999999; pfp_parent = -999999; pfp_numdaughters = -999999; pfp_pdg=-999999; pfp_isprimary=false; clearcosmic = false;
+	  tpc_trk_start_x = DBL_MAX; tpc_trk_start_y = DBL_MAX; tpc_trk_start_z = DBL_MAX;
+	  tpc_trk_end_x = DBL_MAX; tpc_trk_end_y = DBL_MAX; tpc_trk_end_z = DBL_MAX;
+
+//	  std::cout << "starting pointChecker....\n";
+	  pointChecker(*trackList[track_i],start_in_tpc,end_in_tpc,startflag,endflag);
+
+	  //(temporary) adding all valid geo::Point_t object positions to the vectors to be able to identify PCA behavior
+	  geo::Point_t thispt;
+	  size_t nTrackPoints = trackList[track_i]->NPoints();
+	  int midPt = (int)floor(nTrackPoints*0.5);
+	  geo::Point_t midP = trackList[track_i]->LocationAtPoint(midPt);
+
+	  midpt_x = midP.X(); midpt_y = midP.Y(); midpt_z = midP.Z();
+	  track_length = trackList[track_i]->Length();
+
+	  tpc_start_x = trackList[track_i]->StartDirection().X();
```

Comment:
```suggestion
	  tpc_start_dir_x = trackList[track_i]->StartDirection().X();
```
Adding "dir" here avoids confusion with start_pos_x etc.